contracts/RenPool.sol

```diff
@@ -70,16 +77,47 @@ contract RenPool {
         uint senderBalance = balances[sender];
 
         require(senderBalance > 0 && senderBalance >= _amount, "Insufficient funds");
-        require(isLocked == false, "Pool is locked");
-
-        renToken.transfer(sender, _amount);
+        require(isLocked == false);
         totalPooled -= _amount;
         balances[sender] -= _amount;
-
+        renToken.transfer(sender, _amount);
+            
         emit RenWithdrawal(sender, _amount);
+
+    }
+
+    function withdrawRequest(uint _amount) external {
+        address sender = msg.sender;
+        uint senderBalance = balances[sender];
+
+        require(senderBalance > 0 && senderBalance >= _amount, "Insufficient funds");
+        require(isLocked == true, "The pool is not locked, please do a regular withdraw");
+        withdrawRequests[sender] = WithdrawRequest(msg.sender, _amount);
+
     }
 
     function balanceOf(address _addr) external view returns(uint) {
         return balances[_addr];
     }
+
+    function fullfillWithdrawRequest(address _withdrawRequestAddress) external {
+        require(isLocked == true);
+
+        WithdrawRequest memory withdrawRequest = withdrawRequests[_withdrawRequestAddress];
+
+        require(renToken.transferFrom(msg.sender, address(this), withdrawRequest.amount));
+
+
+        // Transfering the balance
+        balances[msg.sender] += withdrawRequest.amount;
+        balances[withdrawRequest.user] -= withdrawRequest.amount;
+
+    
+        // withdraw funds
+        renToken.transfer(withdrawRequest.user, withdrawRequest.amount);
+
```

Comment:
Let's introduce some variables here, we are repeating `withdrawRequest.amount` and `withdrawRequest.user` several times